app/controllers/reactions_controller.rb

```diff
@@ -0,0 +1,34 @@
+class ReactionsController < ApplicationController
+  before_action :set_variables
+  def like
+    if @reaction.blank?
+      @reaction = Reaction.create(user: current_user, post: @post, reaction_type: params[:reaction_type])
+      @reaction.save!
+    elsif @reaction.reaction_type == true
+      @reaction.update(reaction_type: nil)
+    else
+      @reaction.update(reaction_type: true)
+    end
+    redirect_to root_path
```

Comment:
need to update the if else here use find_or_initialize_by here

Reply:
find_or_initialize_by breaks my code logic that is why i did not use it

Reply:
the logic you applied the other day is not working